### PR TITLE
[release-3.11] Do not require a LDAP CA cert if one is not defined for LDAP IdP

### PR DESCRIPTION
**Pull request by Conan-Kudo**
It is not required to define a CA certificate for LDAP in order to use
it securely. Thus, in this scenario, one should not be requested.

This is also needed for openshift-ansible in the 3.10 branch to fix it there.

### CODE REVIEW

**Comment by vrutkovs** [comment]
>It is not required to define a CA certificate for LDAP in order to use
it securely

It may be required for some LDAP servers. Also, this change would mask the fact that container's CA bundle is used

**Comment by Conan-Kudo** [comment]
@vrutkovs That is the intent of this change, though. I _want_ to use the CA bundle.

**Comment by Conan-Kudo** [comment]
@vrutkovs Moreover, using a CA cert file is explicitly optional because it can use the CA cert bundle.

**Comment by vrutkovs** [comment]
In this case you must specify which CA bundle would be used. 
Either specify a local CA bundle via `openshift_master_ldap_ca_file` or place it in `/etc/origin/master/{}_ldap_ca.crt'.format(self.name)` (as different identity providers may use different CA bundles).
python script would ensure `ca` field would be set, user-provided `ca` option is ignored

See https://docs.openshift.com/container-platform/3.11/install_config/configuring_authentication.html#identity-providers-ansible for more information

**Comment by Conan-Kudo** [comment]
@vrutkovs I just tried this, and that doesn't work.

**Comment by Conan-Kudo** [comment]
Wait, I messed up, let me try again...

**Comment by Conan-Kudo** [comment]
Okay, that works...
